src/Orchard.Web/Modules/Orchard.Users/Module.cs

```diff
@@ -60,13 +62,19 @@ public override void ConfigureServices(IServiceCollection serviceCollection)
 
             serviceCollection.TryAddScoped<IUserStore<User>, UserStore>();
 
+            var dataProtectionProvider = DataProtectionProvider.Create(_tenantName);
             serviceCollection.Configure<IdentityOptions>(options =>
             {
                 options.Cookies.ApplicationCookie.CookieName = "orchauth_" + _tenantName;
                 options.Cookies.ApplicationCookie.CookiePath = _tenantPrefix;
                 options.Cookies.ApplicationCookie.LoginPath = new PathString("/Orchard.Users/Account/Login/");
                 options.Cookies.ApplicationCookie.AccessDeniedPath = new PathString("/Orchard.Users/Account/Login/");
+                options.Cookies.ApplicationCookie.DataProtectionProvider = dataProtectionProvider;
+                options.Cookies.ExternalCookie.DataProtectionProvider = dataProtectionProvider;
+                options.Cookies.TwoFactorRememberMeCookie.DataProtectionProvider = dataProtectionProvider;
+                options.Cookies.TwoFactorUserIdCookie.DataProtectionProvider = dataProtectionProvider;                
             });
```

Comment:
Using a different DataProtectionProvider per tenant I expect it is enough to have isolation between tenants at cookie level. At least I have checked it doesn't breaks anything.